FEATURE: extend widget-dropdown to accept disabled option

lis2 opened the pull request:
The widget should accept the disabled option.
In that case, CSS class "disabled".
In addition, after click dropdown will not be shown.

jjaffeux commented on app/assets/javascripts/discourse/app/widgets/widget-dropdown.js:
```diff
@@ -199,15 +200,19 @@ export const WidgetDropdownClass = {
   buildClasses(attrs) {
     const classes = ["widget-dropdown"];
     classes.push(this.state.opened ? "opened" : "closed");
+    if (this.state.disabled) {
```
can we keep the same style as the line above please? (I don't mind this style, but I would prefer consistency).

lis2 commented:
@jjaffeux thank you for review. I added one more thing. I changed strategy from "fixed" to "absolute". It is solving weird bug I experienced. Sometimes when you are unlucky, dropdown is in a wrong place (popper calculation is incorrect). With absolute it works fine.

<img width="505" alt="Screen Shot 2021-05-26 at 11 43 26 am" src="https://user-images.githubusercontent.com/72780/119590755-bb8cee00-be18-11eb-8b57-09881a02b933.png">

What do you think about changing that strategy?

jjaffeux commented:
Yes we can go with this, I will eventually go with details/summary here too anyways
